builder/vsphere/clone/step_add_cdrom.go

```diff
@@ -0,0 +1,59 @@
+//go:generate struct-markdown
+//go:generate mapstructure-to-hcl2 -type CDRomConfig
+
+package clone
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type CDRomConfig struct {
+	// Which controller to use. Example: `sata`. Defaults to `ide`.
+	CdromType string `mapstructure:"cdrom_type"`
+}
+
+type StepAddCDRom struct {
+	Config *CDRomConfig
+}
+
+func (c *CDRomConfig) Prepare() []error {
+	var errs []error
+
+	if c.CdromType != "" && c.CdromType != "ide" && c.CdromType != "sata" {
+		errs = append(errs, fmt.Errorf("'cdrom_type' must be 'ide' or 'sata'"))
+	}
+
+	return errs
+}
+
+func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	vm := state.Get("vm").(driver.VirtualMachine)
+
+	if s.Config.CdromType == "sata" {
+		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
```

Comment:
here, if there's another error than the `driver.ErrNoSataController` should we say something ?

Reply:
Not sure why this was implemented this way ignoring any other error, this is how is implemented in vsphere-iso. Maybe is a bug 🤔  but maybe is not and  ErrNoSataController is the only valid error here. I'll look the API code and look for other possible errors.